### ISSUE TITLE
Updating weights gradually after a new token has been added will fail

### ISSUE THREAD

**Issue by samczsun** [balancer-labs/configurable-rights-pool#15]
https://github.com/balancer-labs/configurable-rights-pool/blob/be02d74740add29aed928e8dc598356f0181f961/contracts/ConfigurableRightsPool.sol#L566-L568

This is because the length of `_startWeights` is the original number of tokens, while `poolTokens.length` might be larger if a new token was added. If weights are gradually updated after a new token is added, this loop will revert due to the out of bounds access

**Comment by EndymionJkb** [comment]
Right - it says "invalid opcode." And it's actually worse than that! Losing synchronization after an add is simple enough to fix - I'm now appending newToken.denorm to _startWeights after applying an add.

But we'll have an analogous problem after a remove. It won't "fail," but it won't work correctly. When you delete a token in BPool, it swaps the token being deleted with the one on the end, then deletes the last one. So the _startWeights array will be *longer* than the underlying pool array.

I fixed it by resyncing after a removeToken. It does a pop() to shorten _startWeights, then sets _startWeights[i] in a loop from bPool.getCurrentTokens(), ensuring everything is aligned.

It does mean that callers of updateWeightsGradually need to be sure they're passing in weights in the right order, if they've been adding and removing. So I documented that in updateWeightsGradually, and described how to retrieve the token list from the underlying pool in case there was ever any confusion.

And I've added tests for calling updateWeightsGradually after both an add and a remove - additionally checking for alignment of the final weights after the update is complete.
